kong/conf_loader.lua

```diff
@@ -514,6 +539,48 @@ local function load(path, custom_conf)
   -- Merging & validation
   -----------------------
 
+  -- find dynamic keys that need to be loaded
+  do
+    local dynamic_keys = {}
+    local function find_dynamic_keys(t, prefix)
+      if not t then
+        return
+      end
+
+      for k, v in pairs(t) do
+        local _, _, directive = string.find(k, "(" .. prefix .. ".+)")
+        if directive then
+          dynamic_keys[directive] = true
+          if tonumber(v) then
+            t[k] = string.format("%q", v)
+          end
+        end
+      end
+    end
+
+    local env_vars, err = env.read_all()
+    if err then
+      return nil, err
+    end
+
+    local kong_env_vars = {}
+    for k, v in pairs(env_vars) do
+      local clean_k = string.lower(k)
+      local kong_var = string.match(clean_k, "^kong_(.+)")
+      if kong_var then
+        kong_env_vars[kong_var] = true
+      end
+    end
+
+    for _, prefix in pairs(CONF_KEY_PREFIXES) do
+      find_dynamic_keys(custom_conf, prefix)
+      find_dynamic_keys(kong_env_vars, prefix)
+      find_dynamic_keys(from_file_conf, prefix)
+    end
+
+    defaults = tablex.merge(defaults, dynamic_keys, true)
```

Comment:
Can you add a comment that this is building the union? This way the intent is clearer (as discussed) to readers who see `tablex.merge()` usage several times in this module. See below:
```lua
conf = tablex.merge(conf, defaults) -- intersection (remove extraneous properties)
```